### ISSUE TITLE
Different seeds lead to very different results in unsup  BERT

### ISSUE THREAD

**Issue by kongds** [princeton-nlp/SimCSE#63]
Hello, thank you for sharing code. 

I found the result of unsup BERT seems unstable to different seeds.  I can reproduce the result of 76~ with the default seed, but when changing to different the seeds, the results seems worse than default seed. (I use '--seed' flag to set seed)
![image](https://user-images.githubusercontent.com/19282154/126934638-74e7e770-7033-4583-8750-fb4ab4b5d93d.png)

Have you encountered this problem?

**Comment by gaotianyu1350** [comment]
Hi,

For the unsupervised model, the performance could be affected by different random seeds, hardwares, validation steps, etc. We have been using the default seed of Huggingface's transformers.

**Comment by kongds** [comment]
Thank you for your answer.
In this case, the performance is only affected by seeds. But I still have some concerns:
1) Seeds may have a deviation of about 2% in spearsman, using a single seed as the final result seems inaccurate.
2) When we use no default seeds, the performance seems to be worse. Does the searched hyperparameter only apply to the default seed?

**Comment by gaotianyu1350** [comment]
Hi,

Thanks for the suggestion. We are also planning to add more trials and take the average for more accurate results. It's indeed probably the case that you need to re-tune the hyper-parameters when changing the seeds.

**Comment by kongds** [comment]
Thank you
